internal/datastore/context.go

```diff
@@ -47,6 +47,14 @@ func (p *ctxProxy) ReadWriteTx(
 	return p.delegate.ReadWriteTx(ctx, f, opts...)
 }
 
+func (p *ctxProxy) IsStrictReadModeEnabled() bool {
+	if srm, ok := p.delegate.(datastore.StrictReadDatastore); ok {
```

Comment:
This would work only if the direct delegate implements the interface. We know this has regressed in the past due to the addition of new proxies to the chain.

This is what we built [`UnwrapAs`](https://github.com/authzed/spicedb/blob/f42cc3fe4ee403e2b60b63dc3867569ccdb75927/pkg/datastore/datastore.go#L641) for, I'd recommend using it.

Reply:
I actually did this explicitly: I wanted this proxy to be *only* given the raw datastore, rather than any wrapped ones, to ensure it was immediately above the datastore itself (which is necessary for safety). Shall I add a comment?

Reply:
After discussion, changed